tests/plugins/test_amp_plugins.py

```diff
@@ -99,6 +99,52 @@ def test_amp_gradient_unscale(tmpdir, accum: int):
     trainer.fit(model)
 
 
+@RunIf(min_gpus=1, amp_native=True)
+def test_amp_skip_optimizer(tmpdir):
+    """
+    Test that optimizers can be skipped when using amp
+    """
+
+    class CustomBoringModel(LightningModule):
```

Comment:
you can inherit from BoringModel, then you don't need the datamodule.

Reply:
Thanks for the tip! Updated the PR accordingly.